### ISSUE TITLE
VS-Code stutter (delayed update/draw calls?)

### ISSUE THREAD

**Issue by Azayel** [prasanthrangan/hyprdots#496]
Hello There! 

I recently abandoned Windows and switched completely to Linux and managed to fix most of my problems my self. I do not know if i am the origin of this problem, because I installed some of my favorite programs or this is a issue with hyprland itself. 

I am currently facing an weird problem with VS-Code and other Electron(?) based applications. I know there is a bad compatibility with wayland and electron, but seeing that code is majorly used here and got its own shortcut, I presume that this problem is on my end.

It appears that code has some kind of delayed draw call(?) update call(?) for user input. The application itself does not flicker, but user input is really weird. Typing is not smooth. During a typing session, the last written letters randomly disappear and appear as if the program can not keep up with the input and lags behind. Moving the mouse over explorer elements and settings is the same. Usually if you hover over some kind of folder or file it gets highlighted but moving the mouse the highlight randomly appears over past positions. (Attached Image Hovering over Shader.h but highlight is at mainloop.h). Hovering over the window settings reveals a black window but moving the mouse over it shows the settings again. Onetime I was scrolling through the code settings the text overlapped with the previous "scroll frame".

I do not know if this helps but some people said using these flags is the only option.
I tried adding these:
` '"--enable-features=UseOzonePlatform --ozone-platform=wayland"`
But was given this output:

```
Warning: 'enable-features' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'ozone-platform' is not in the list of known options, but still passed to Electron/Chromium.

```
My Specs: 

- AMD Ryzen 5 5600X (12) @ 3.700GHz 
- NVIDIA GeForce RTX 3080 12GB 
- GPU Driver : NVIDIA 545.29.02
- Memory : 2626MiB / 15901MiB 

Is this a nvidia Issue or am I the culprit who is responsible for this bug :)? I mean

![231115_14h53m04s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/80709673/6250b7e8-2e55-493f-bff2-7098770682fd)

![vscode](https://github.com/prasanthrangan/hyprdots/assets/80709673/9b45342c-7b46-464f-8c9f-b0497a1ff9bd)

**Comment by zer0cha0s** [comment]
Hi @Azayel, i ran into this problem yesterday along with some major screen tearing playing video games. 
I'm pretty sure it was the 545.29.02 NVIDIA drivers causing the issue. 
I wasn't able to find a fix for this so rolled back my drivers to 535.113.01 and everything is fine for me now.

**Comment by Azayel** [comment]
Hey Thank You for replying @zer0cha0s, sadly I think it wont be possible for me to re-roll back, due to not having a snapshot of the given NVIDIA driver version... Meaning I need to do it manually, which is a really bad idea. Given this thread, I even think I need to change my Kernel Version, do I?

[https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia/-/commits/main](url) 

If this issue lies truly in the drivers, than it looks like I have to wait for an update.

**Comment by imhatep** [comment]
> Hi @Azayel, i ran into this problem yesterday along with some major screen tearing playing video games. I'm pretty sure it was the 545.29.02 NVIDIA drivers causing the issue. I wasn't able to find a fix for this so rolled back my drivers to 535.113.01 and everything is fine for me now.

yea, i swap to rx580 temporarily instead of downgrading)
But it's not important, on my 1660 there are frequent bugs on Wayland with 5xx drivers

**Comment by Azayel** [comment]
Well it is not that bad I guess. Will embark upon the path of waiting, for another driver update... Downgrade did not really solve the issue in code. Well nvim it is! But for java I think I would need to switch to intelij in the meantime... Yikes. 

Thanks for the replies!

**Comment by prasanthrangan** [comment]
Hey @Azayel , can you please try if any of these options works for you?

```
code --disable-gpu-sandbox
```
or
```
code --disable-gpu
```

**Comment by Azayel** [comment]
Hello @prasanthrangan, thaks for the reply.

`code --disable-gpu-sandbox` did not fix the problem but

`code --disable-gpu` surprisingly did fix the problem.

Thanks!

**Comment by prasanthrangan** [comment]
great thanks, I'll close this!

**Comment by t101804** [comment]
> Hi @Azayel, i ran into this problem yesterday along with some major screen tearing playing video games. I'm pretty sure it was the 545.29.02 NVIDIA drivers causing the issue. I wasn't able to find a fix for this so rolled back my drivers to 535.113.01 and everything is fine for me now.

hey its also working for me , but isee nvidia upgrade their pkg. its 550 is no stutering?

**Comment by willlogs** [comment]
550.78 Same problem.
Downgrading to 535.113.01 fixed it.
